Review Only

LesliePajuelo opened the pull request.

McCloudFace committed:
initial commit

McCloudFace committed:
cleanup

McCloudFace committed:
initial schema

McCloudFace committed:
add example-data

McCloudFace committed:
add example data, refactor schema.sql

McCloudFace committed:
testing new branch

rybrennan committed:
Merge pull request #3 from rpt09-scully/testing

testing new branch

McCloudFace committed:
db setup on heroku, json test responding

rybrennan committed:
Merge pull request #4 from rpt09-scully/herokuSetup

mvp test for heroku

McCloudFace committed:
[wip] api endpoints
  - add getAtrail, getTags functions in db

McCloudFace committed:
get /:trailId/trailInfo functioning as expected

rybrennan committed:
Merge pull request #5 from rpt09-scully/queries

implement /:trailId/trailInfo api endpoint

cjm771 committed:
[Edit] .gitignore, .readme dev setup, port 3000 -> 3001

McCloudFace committed:
Merge pull request #6 from rpt09-scully/pr/readme-dev-setup-and-misc

[Edit] .gitignore, .readme dev setup, port 3000 -> 3001

McCloudFace committed:
cleanup

McCloudFace committed:
add generateSql for generate insert statements

McCloudFace committed:
implement seed-database from package.json script

cjm771 committed:
Merge pull request #7 from rpt09-scully/generateData

Implement Seed Database

McCloudFace committed:
add static route

McCloudFace committed:
[wip] adding test suite

McCloudFace committed:
[wip] testing tasks
  - add mocha, chai, tests
  - comment db functions & db tests
  - add client/dist to repo

McCloudFace committed:
implement server app tests

mweber-ak committed:
Merge pull request #8 from rpt09-scully/testSuite

adding test suite (plz merge when ready)

LesliePajuelo commented on database/index.js:
```diff
@@ -0,0 +1,56 @@
+  }
+});
+
+var getAtrail = (id, callback) => {
```
giving the callback a more meaningful name will be helpful in debugging.

LesliePajuelo commented on database/index.js:
```diff
@@ -0,0 +1,56 @@
+  knex.select('tag_name').from('tags')
+    .whereIn('tag_id', subquery)
+    .then((tags) => {
+      var tagsArr = [];
```
Nitpicky naming: tagsArr? When this is being rendered on the front-end, what is that section called?

LesliePajuelo commented on database/index.js:
```diff
@@ -0,0 +1,56 @@
+};
+
+// for testing only
+var getAllTrails = (callback) => {
```
Long term planning: add some protection here. Set a limit of how many records will be queried. 

When this gets picked up by FEC (becomes a production "prototype"), it'll go from 100 records to 10 million

LesliePajuelo commented on generateSql.js:
```diff
@@ -0,0 +1,99 @@
+// This is a utility file to generate SQL INSERT statments
```
Great comment!

LesliePajuelo commented on generateSql.js:
```diff
@@ -0,0 +1,99 @@
+
+// INSERT INTO trail_tags (trail_id, tag_id) VALUES (21, 8);
+
+const trailTags = function () {
```
Nitpicky naming: 😿 single letter variable names. str, trTag (that is tr???) 
Good function name tho, makes it possible to not have to read this. Other than, wait, I think trailTags means ... trail tags, it really means insert query for trail tags.

LesliePajuelo commented on generateSql.js:
```diff
@@ -0,0 +1,99 @@
+// INSERT INTO trail_tags (trail_id, tag_id) VALUES (21, 8);
+
+const trailTags = function () {
+  let str = '';
```
putting aside that trTags is not a great name, why not simply set trTag = '' and accumulate into it?

what's the benefit of the separation?

LesliePajuelo commented on package.json:
```diff
@@ -1,17 +1,30 @@
-  "author": "",
-  "license": "",
+  "name": "ninetrails-trail-service",
+  "version": "0.1.0",
```
Continue to update version as you add features/fixes (breaking.feature.fix)

LesliePajuelo commented on server/index.js:
```diff
@@ -0,0 +1,62 @@
+  Example: http://localhost:3001/3/trailInfo
+  for the Trail where trail_id === 3
+*/
+app.get('/:trailId/trailInfo', (req, res) => {
```
Curiosity, why `/:trailId/trailInfo' instead of '/trailInfo/:trailId'?

McCloudFace replied:
When the team met originally to assign services we settled on that structure for endpoints. I believe those are all documented in the planning doc [here](https://www.lucidchart.com/documents/edit/a54b4ef2-2758-4eeb-bba4-b5d5c2bdcf03/0?shared=true&).

LesliePajuelo commented on server/index.js:
```diff
@@ -0,0 +1,62 @@
+*/
+app.get('/:trailId/trailInfo', (req, res) => {
+  var theId = req.params.trailId;
+  db.getAtrail(theId, (row) => {
```
Pulling out this anonymous function into a named function will make it very easy to unit test and be reusable.

LesliePajuelo commented on test/db.test.js:
```diff
@@ -0,0 +1,48 @@
+   context. If you do not need to use Mocha’s context, lambdas
+   should work. However, the result will be more difficult to
+   refactor if the need eventually arises.
+ *
```
🎆 🎉 For the very helpful comment

LesliePajuelo commented on test/db.test.js:
```diff
@@ -0,0 +1,48 @@
+      });
+    });
+  });
+});
```
Also test if the values are as expected. 

There's no type checking in other areas of your app, so at the very least you'd want to test that trail_name isn't an array for instance.

LesliePajuelo commented on test/db.test.js:
```diff
@@ -0,0 +1,48 @@
+    });
+  });
+});
+
```
this is prob all the testing you need for getTags()

However, for getAtrail() you're shaping data in the callback. Please inlcude a unit test (prob different file) for that shaping

McCloudFace committed:
Update readme per template

McCloudFace committed:
Edit and update readme

McCloudFace committed:
db test file refactored to Jest
  - both tests passing

McCloudFace committed:
Tests migrated to Jest
  - Installed babel, React & webpack

MengRS committed:
Merge pull request #11 from rpt09-scully/readme-template

Please merge: Update readme per template

cjm771 committed:
Merge pull request #14 from rpt09-scully/mocha-to-jest

Mocha to jest

McCloudFace committed:
update package.json scripts

McCloudFace committed:
Set up mvp React

McCloudFace committed:
update log

McCloudFace committed:
Render placeholder content in trail component

McCloudFace committed:
feat: set up Bootstrap as css module

McCloudFace committed:
feat: add namespace and break out separate components

McCloudFace committed:
fix: revert name spacing

MengRS committed:
Update README.md

MengRS committed:
Merge pull request #17 from rpt09-scully/client-setup

[Review & merge]: Client setup

Reviewed and merged.

McCloudFace committed:
fix: implement NT name space

McCloudFace committed:
feat: [wip] static state data

McCloudFace committed:
fix: response id and getTrail rename

McCloudFace committed:
fix getTrail test

cjm771 committed:
Merge pull request #26 from rpt09-scully/data-fix

[review & merge] fix: response id and getTrail rename

rybrennan committed:
Merge pull request #24 from rpt09-scully/02_client_setup

[review & merge] 02 client setup

McCloudFace committed:
update component layout

McCloudFace committed:
style: [wip] classes in service layout

McCloudFace committed:
[wip] add component structure

McCloudFace committed:
[wip] position heroStats

McCloudFace committed:
feat: add cors

MengRS committed:
Merge pull request #28 from rpt09-scully/cors

[plz merge] add cors

rybrennan committed:
Merge pull request #27 from rpt09-scully/service-layout

[MERGED] Service layout

McCloudFace committed:
[wip] styles & components

McCloudFace committed:
render trail data from api

McCloudFace committed:
[wip] style traildescription

cjm771 committed:
Merge pull request #29 from rpt09-scully/02_client-setup

[plz merge asap] 02 client setup

McCloudFace committed:
public.app

McCloudFace committed:
[Updated: should be fixed. please squash commits to latest when merging] Component endpoints (#30)

* fix: refactor trailInfo endpoint

* feat: [wip] add banner endpoint

* feat: [wip] add trailDescription endpoint

* fix: reinstall axios, add nodemon

* fix: [empty] refactor axios in serviceHelpers

  - add axios multiple concurrent requests for better promise handling

* fix: refactor serviceHelpers

  - functioning as expected now

McCloudFace committed:
merge: component-endpoints

  - Merge branch 'master' of github-mcc:rpt09-scully/trail-service
  - banner and trailDescription endpoints implemented

McCloudFace committed:
[review & merge w squash commit] feat: respond to trailId in URL (#32)

* public.app

* feat: add state endpoints to both components

  - each component making separate api calls for pre-shaped responses

* feat: components render async from new endpoints

  - Add axios requests for component-specific endpoints
  - Refactor to React "Update on Async Rendering" format
  - https://jffc.us/react-async-external-data

* feat: respond to trailId in URL

McCloudFace committed:
fix: merge conflict public/app.js

McCloudFace committed:
feat: [wip] initial aws deploy

McCloudFace committed:
feat: deploy trail service to AWS

 - deployed database to RDS, service to EB

 - trail-service API functioning as expected:
 http://trail-env.8jhbbn2nrv.us-west-2.elasticbeanstalk.com/3/trailInfo

 - trail-service deployed (single widget)
 http://trail-env.8jhbbn2nrv.us-west-2.elasticbeanstalk.com/3

 - internal APIs for components functioning as expected
 http://trail-env.8jhbbn2nrv.us-west-2.elasticbeanstalk.com/3/banner
 http://trail-env ... /3/trailDescription

  - Add webpack environment option to new scripts of package.json
  - Add check for environment in components
  - [wip] cause photo service is down rn

McCloudFace committed:
feat: widget view renders in dev and prod

McCloudFace committed:
chore: run react-prod for deploy

McCloudFace committed:
fix: cors as reg dependency

McCloudFace committed:
fix: banner css

rybrennan committed:
Merge pull request #34 from rpt09-scully/deploy-first

[MERGED] Deploy first

McCloudFace committed:
chore: deploy update

McCloudFace committed:
fix: service widget page css

McCloudFace committed:
Merge pull request #35 from rpt09-scully/3001-css

[plz merge] css tweaks for service widget page

McCloudFace committed:
fonts

McCloudFace committed:
Merge pull request #36 from rpt09-scully/css-tweaks

fonts

McCloudFace committed:
readme

McCloudFace committed:
client tweak
